index.ts

```diff
@@ -196,8 +197,21 @@ function isNodeKindModuleDeclaration(value: ts.Node): value is ts.ModuleDeclarat
 
 export default function generate(options: Options): Promise<void> {
 
-	if (Boolean(options.main) && !options.name) {
-		throw new Error('name is required when main is used');
+	if (Boolean(options.main) !== Boolean(options.name)) {
+		if (Boolean(options.name)) {
+      // since options.name used to do double duty as the prefix, let's be
+      // considerate and point out that name should be replaced with prefix.
+      // TODO update this error message when we finalize which version this change
+      // will be released in.
+			throw new Error(
+				`name and main must be used together.  Perhaps you want prefix instead of name?
+				In version 2.1, name did double duty as the option to use to prefix module names
```

Comment:
May want to be explicit that we mean dts-generator version rather than TypeScript version number given that the version numbers are almost the same?